packages/react-storage/src/components/StorageBrowser/controls/hooks/useDataTable.ts

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import {
+  DataTableButtonDataCell,
+  DataTableDateDataCell,
+  DataTableTextDataCell,
+  DataTableDataCell,
+  DataTableProps,
+  SortDirection,
+  DataTableNumberDataCell,
+} from '../../composables/DataTable';
+import { useControlsContext } from '../context';
+import { compareButtonData } from '../compareFunctions/compareButtonData';
+import { compareDateData } from '../compareFunctions/compareDateData';
+import { compareNumberData } from '../compareFunctions/compareNumberData';
+import { compareTextData } from '../compareFunctions/compareTextData';
+
+export type UseDataTable = () => {
+  props?: DataTableProps;
+};
+
+interface SortState {
```

Comment:
Sorting is now encapsulated entirely within a DataTable

Reply:
Thank you 🙏